tests/sentry/seer/similarity/test_utils.py

```diff
@@ -1051,3 +1052,135 @@ def test_ignores_events_not_grouped_on_stacktrace(self) -> None:
+
+        simple_event = Event(
+            event_id="12312012041520130908201311212012",
+            project_id=self.project.id,
+            data={
+                "title": "Simple error",
+                "platform": "python",
+                "stacktrace_string": simple_stacktrace,
+            },
+        )
+
+        complex_event = Event(
+            event_id="12312012041520130908201311212012",
+            project_id=self.project.id,
+            data={
+                "title": "Complex error",
+                "platform": "python",
+                "stacktrace_string": complex_stacktrace,
+            },
+        )
+
+        simple_count = get_token_count(simple_event, None, "python")
+        complex_count = get_token_count(complex_event, None, "python")
+
+        # Exact token counts for these specific strings
+        assert simple_count == 18
+        assert complex_count == 116
+
+    def test_calculates_with_provided_variants(self) -> None:
+        """Test that get_token_count uses provided variants to calculate stacktrace."""
+        variants = self.event.get_grouping_variants(normalize_stacktraces=True)
+
+        token_count = get_token_count(self.event, variants, "python")
+
+        # Exact token count for the test event's stacktrace
+        assert token_count == 28
+
+    def test_returns_zero_for_empty_stacktrace(self) -> None:
+        """Test that get_token_count returns 0 for events with no meaningful stacktrace."""
+        # Create an event with no stacktrace data
+        empty_event = Event(
+            event_id="12312012041520130908201311212012",
+            project_id=self.project.id,
+            data={
+                "title": "Empty event",
+                "platform": "python",
+            },
+        )
+
+        variants = empty_event.get_grouping_variants(normalize_stacktraces=True)
+        token_count = get_token_count(empty_event, variants, "python")
+
+        assert token_count == 0
+
+    def test_handles_exception_gracefully(self) -> None:
+        """Test that get_token_count handles exceptions gracefully and returns 0."""
+        # Create an event with cached stacktrace that will cause tiktoken to fail
+        broken_event = Event(
+            event_id="12312012041520130908201311212012",
+            project_id=self.project.id,
+            data={
+                "title": "Broken event",
+                "stacktrace_string": "valid stacktrace",  # Will cause encoding to be called
+            },
+        )
+
+        # Mock tiktoken encoding to raise an exception
+        with patch("sentry.seer.similarity.utils.TIKTOKEN_ENCODING.encode") as mock_encode:
+            mock_encode.side_effect = ValueError("Tiktoken encoding failed")
+
+            with patch("sentry.seer.similarity.utils.logger.error") as mock_logger_error:
+                token_count = get_token_count(broken_event, variants=None, platform="python")
```

Comment:
```suggestion
                token_count = get_token_count(broken_event, variants={}, platform="python")
```
(Since `variants` should be non-nullable.)